### ISSUE TITLE
Write the .d.ts as `declare module ${pathToFile}`

### ISSUE THREAD

**Issue by SrBrahma** [Quramy/typed-css-modules#270]
The current implementation doesn't allow to import the .css files using absolute import.

To support it, the .d.ts would need to do 

```ts
declare module 'path/to/your/cssfile.css' {
  const styles: {
    readonly "drawer": string;
    readonly "overlay": string;
    // ... other class names
  };
  export = styles;
}
```

instead of 

```ts
declare const styles: {
  readonly "drawer": string;
  readonly "overlay": string;
  ...
};
export = styles;
```

**Comment by ryami333** [comment]
> The suggestion also allows all of the types to be in a single file

Yes, or in multiple files but in a single dedicated directory like `/types`. We would really like this feature ❤️

**Comment by stevenpetryk** [comment]
As far as I know this simply doesn't work in TypeScript. Ambient modules cannot declare the types for specific files, only general imports. So for example,

```ts
declare module "test.module.css" {
  const styles: {
    readonly foo: string
    readonly background: string
  }
  export = styles
}
```

Will only add typings for:

```ts
import styles from "test.module.css";
```

Not the much more common:

```ts
import styles from "./test.module.css";
```

**Comment by ryami333** [comment]
@stevenpetryk most of what you said is correct - except one little detail that actually makes a workaround possible.

> you are not allowed to use relative or absolute paths

You're right that relative paths don't work, but _absolute paths do_. Given that, consider a project like this:

<img width="158" alt="Screenshot 2024-03-12 at 19 32 56" src="https://github.com/Quramy/typed-css-modules/assets/12481532/f84c5fdb-a47e-4f1d-8095-e3816d5db65a">

With tsconfig's [path](https://www.typescriptlang.org/tsconfig#paths) aliases, you can add an import alias for any folder, including the root folder. You could alias the `src/styles` folder as `@styles` with this Tsconfig:

```js
{
  // …
  "compilerOptions": {
    // …
    "paths": {
      "@styles/*": ["src/styles/*"]
    }
  }
}
```

Now we can do this (although you'd still get a `Cannot find module '@styles/foo.module.css' or its corresponding type declarations.` error for now): 

```ts
import fooStyles from '@styles/foo.module.css`;
```

Now, you can add definition file anywhere in the project, like `src/global.d.ts` and declare modules using the aliased path:

```ts
// global.d.ts

declare module "@styles/foo.module.css" {
  const stylesheet: { hello: "world" }; // <- to be generated by TCM
  export default stylesheet;
}

declare module "@styles/bar.module.css" {
  const stylesheet: any; // <- to be generated by TCM
  export default stylesheet;
}
```

And now the `foo` import will be typed as `{ hello: "world" }`.

<img width="655" alt="Screenshot 2024-03-12 at 19 41 22" src="https://github.com/Quramy/typed-css-modules/assets/12481532/9d81f1ac-2f5b-4c26-94cf-b69f39224c86">

**Comment by stevenpetryk** [comment]
Hm, yeah I suppose that works. Could be worth it if you're willing to rewrite all your existing CSS imports.
